host_source.go

```diff
@@ -599,44 +599,40 @@ func (r *ringDescriber) GetHosts() ([]*HostInfo, string, error) {
 }
 
 // Given an ip/port return HostInfo for the specified ip/port
-func (r *ringDescriber) getHostInfo(ip net.IP, port int) (*HostInfo, error) {
+func (r *ringDescriber) getHostInfo(hostID UUID) (*HostInfo, error) {
 	var host *HostInfo
-	iter := r.session.control.withConnHost(func(ch *connHost) *Iter {
-		if ch.host.ConnectAddress().Equal(ip) {
-			host = ch.host
-			return nil
-		}
-
-		return ch.conn.query(context.TODO(), "SELECT * FROM system.peers")
-	})
+	for _, table := range []string{"system.peers", "system.local"} {
+		iter := r.session.control.withConnHost(func(ch *connHost) *Iter {
+			if ch.host.HostID() == hostID.String() {
+				host = ch.host
+				return nil
+			}
 
-	if iter != nil {
-		rows, err := iter.SliceMap()
-		if err != nil {
-			return nil, err
-		}
+			return ch.conn.query(context.TODO(), fmt.Sprintf("SELECT * FROM %s", table))
+		})
 
-		for _, row := range rows {
-			h, err := r.session.hostInfoFromMap(row, &HostInfo{port: port})
+		if iter != nil {
+			rows, err := iter.SliceMap()
 			if err != nil {
 				return nil, err
 			}
 
-			if h.ConnectAddress().Equal(ip) {
-				host = h
-				break
-			}
-		}
+			for _, row := range rows {
+				h, err := r.session.hostInfoFromMap(row, &HostInfo{port: r.session.cfg.Port})
+				if err != nil {
+					return nil, err
+				}
 
-		if host == nil {
-			return nil, errors.New("host not found in peers table")
+				if h.HostID() == hostID.String() {
+					host = h
+					break
+				}
+			}
 		}
 	}
 
 	if host == nil {
 		return nil, errors.New("unable to fetch host info: invalid control connection")
-	} else if host.invalidConnectAddr() {
-		return nil, fmt.Errorf("host ConnectAddress invalid ip=%v: %v", ip, host)
```

Comment:
Maybe the function shall be changed not a call removed?

Reply:
i don't remember why i removed it, probably by mistake. Brought it back and all tests are passing. Good catch